dsc/db/models.py

```diff
@@ -0,0 +1,145 @@
+
+
+class OptionalItemAttributes(TypedDict, total=False):
+    dspace_handle: str
+    status: str
+    status_details: str
+    ingest_date: str
+    last_submission_message: str
+    last_result_message: str
+    last_run_date: str
+    submit_attempts: int
+    ingest_attempts: int
+
+
+class ItemSubmissionDB(Model):
+    """A DynamoDB model representing an item submission.
+
+    This model stores information about the state of an item submission
+    as it progresses through the DSC workflow. The table uses a
+    composite primary key consisting of 'batch_id' (partition key)
+    and item_identifier (sort key).
+
+    Attributes:
+        batch_id [partition key]: A unique identifier for the workflow run,
+            also used as an S3 prefix for workflow run files.
+        item_identifier [sort key]: A unique identifier for an item submission
+            in a batch.
+        workflow_name: The name of the DSC workflow.
+        dspace_handle: A persistent, globally unique identifier for a digital object
+            in DSpace. The handle is provided in the DSS result message when
+            an item is successfully ingested into DSpace.
+            NOTE: If the item is sent to a submission queue, the handle is
+            NOT provided.
+        status: The current state of an item submission in the DSC workflow.
+            See dsc.db.models.ItemSubmissionStatus for accepted values.
+        status_details: Additional details regarding the status of an item
+            (e.g., error messages).
+        ingest_date: A date representing when an item was successfully ingested
+            into DSpace. In DynamoDB, the date is stored as a string
+            (in ISO 8601 format).
+        last_submission_message: A serialized JSON string of the latest (most recent)
+            submission message composed and sent to the input SQS queue via
+            the submit command.
+        last_result_message: A serialized JSON string of the latest (most recent)
+            result message composed and sent to the output SQS queue for DSC via DSS.
+        last_run_date: A date representing the last time a DSC CLI command was executed
+            on the item. In DynamoDB, the date is stored as a string (in ISO 8601 format).
+        submit_attempts: The number of attempts to send a submission message to the
+            input SQS queue for DSC. This value is only incremented when the DSC
+            submit command is run for an item.
+        ingest_attempts: The number of attempts to ingest an item into DSpace (run DSS).
+            This value is only incremented when the DSC finalize command is run for
+            an item.
+    """
+
+    class Meta:  # noqa: D106
+        table_name = "dsc"
+
+    batch_id = UnicodeAttribute(hash_key=True)
+    item_identifier = UnicodeAttribute(range_key=True)
+    workflow_name = UnicodeAttribute()
+    dspace_handle = UnicodeAttribute(null=True)
+    status = UnicodeAttribute(null=True)
```

Comment:
Noting for posterity what you shared @jonavellecuerdo, that `pynamodb` does not support an enum or list of accepted values.  I think that's okay.